### ISSUE TITLE
Possible get_uptime_df() issue in wavebank

### ISSUE THREAD

**Issue by El-minadero** [niosh-mining/obsplus#111]
I've discovered a potential issue with generating uptime df's with wavebank. It *might* be my code, but I've tested it as much as possible and cannot see where I'm going wrong.

So the idea is to generate a synthetic trace dataset with gaps like so:

```python
"""
Station    component:
            
                2 min         2 min         2 min         2 min         2 min
                perfect data  one channel   one station   one channel   two stations  
                              gone          gone          incomplete    incomplete

(1)         HHE ------------- ------------- ------------- ------------- ------------- 
            HHN ------------- ------------- ------------- ------------- ------------- 
            HHZ ------------- ------------- ------------- ------------- -------------
            
(2)         HHE ------------- -------------               ------------- ----------
            HHN -------------                             ------------- ----------
            HHZ ------------- -------------               ----------    ---------- 
            
(3)         HHE ------------- ------------- ------------- ------------- ----------
            HHN ------------- ------------- ------------- ------------- ----------
            HHZ ------------- ------------- ------------- ------------- ----------
"""

```
for every trace, let the sampling rate be 50 hz, synthetic network be 'test', and station code just be the string equivalent above. [This](https://pastebin.com/Ha2nu2T9) is my attempt at doing it in an ipynb notebook. Ive decided to put it in pastebin to more succinctly document the behavior.

I then save the generated stream like so:
```python
save_dir = 'save_dir/'
for trace in stream:
    station = trace.stats.station
    network = trace.stats.network
    channel = trace.stats.channel
    time    = trace.stats.starttime.isoformat()
    print(trace.stats.npts)
    file_name = '{}{}.{}.{}.{}.{}'.format(save_dir,network,station,channel,time,'mseed')
    trace.write(file_name,format='mseed')
```
and confirm that files are created like so:
```
test.1.HHE.1970-01-01T....
test.1.HHN.1970-01-01T..
...
test.3.HHZ.1970-01-01T..
```
basically 3 stations with 3 components and several utcdatetime isoformat files are created.

Now, when I go to make a WaveBank in this directory, update it, and gets its uptime df, I only see one station listed. station '2', in the df.
```python
bank = WaveBank(directory)
bank.update_index()
df =bank.get_uptime_df()
    def create_seed(row):
        network = row['network']
        station = row['station']
        return network + '.' + station
df['seed'] = df.apply(lambda row: create_seed(row), axis=1)
unique_stations = df['seed'].unique().tolist()
len(unique_stations)
1
print(unique_stations)
['test.2']
```
So, it seems potentially likely that the uptime_df() is not catching all the possible stations in the directory.

**Comment by d-chambers** [comment]
Yes there does seem to be an issue here, I can reproduce it.

**Comment by El-minadero** [comment]
Also, it seems that ```wb.get_availability_df()``` works fine.

**Comment by d-chambers** [comment]
Ok looking at this more the issue only shows up when *some* channels have gaps and *some* channels do not. This is because `wavebank.get_uptime_df` first gets the availability data frame then merges it with the gaps dataframe. The default of `pd.merge` is to do an inner merge, but in this case we need an outer merge. I should have this fixed in a few minutes. I need to merge in #112 first though.
